genesis/src/lib.rs

```diff
@@ -27,12 +25,12 @@ pub static GENESIS_ACCOUNT_ID: Lazy<AccountId> =
     Lazy::new(|| AccountId::new(GENESIS_DOMAIN_ID.clone(), "genesis".parse().expect("Valid")));
 
 /// Genesis transaction
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize)]
 #[repr(transparent)]
 pub struct GenesisTransaction(pub SignedTransaction);
 
 /// [`GenesisNetwork`] contains initial transactions and genesis setup related parameters.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize)]
 pub struct GenesisNetwork {
```

Comment:
we want genesis to be exactly 1 transaction, not a vec

Reply:
you can remove `GenesisNetwork` and just use `GenesisTransaction` everywhere

Reply:
Changed to operate with GenesisTransaction, although it's more convenient to get GenesisNetwork directly at the final step